environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/PrerequisitesDeleteHandler.java

```diff
@@ -0,0 +1,153 @@
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.Setup;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.prerequisite.AzurePrerequisiteDeleteRequest;
+import com.sequenceiq.cloudbreak.cloud.model.prerequisite.EnvironmentPrerequisiteDeleteRequest;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationEvent;
+import com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteFailedEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.environment.parameters.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameters.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameters.dto.ParametersDto;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+
+@Component
+public class PrerequisitesDeleteHandler extends EventSenderAwareHandler<EnvironmentDto> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PrerequisitesDeleteHandler.class);
+
+    private final EnvironmentService environmentService;
+
+    private final CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    private final CloudPlatformConnectors cloudPlatformConnectors;
+
+    protected PrerequisitesDeleteHandler(EventSender eventSender, EnvironmentService environmentService,
+            CredentialToCloudCredentialConverter credentialToCloudCredentialConverter, CloudPlatformConnectors cloudPlatformConnectors) {
+        super(eventSender);
+        this.environmentService = environmentService;
+        this.credentialToCloudCredentialConverter = credentialToCloudCredentialConverter;
+        this.cloudPlatformConnectors = cloudPlatformConnectors;
+    }
+
+    @Override
+    public String selector() {
+        return DELETE_PREREQUISITES_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        environmentService.findEnvironmentById(environmentDto.getId())
+                .ifPresentOrElse(environment -> {
+                            try {
+                                if (AZURE.name().equals(environmentDto.getCloudPlatform())) {
```

Comment:
Please remove the if here and use the composite pattern to determine the specific cloud-api that is needed to delete the prerequisites. Move every azure specific implementation to the cloud-azure module.

Reply:
See PublicKeyCreationHandler for reference.

Reply:
Why is this if still here? Please remove it like in PublicKeyCreationHandler

Reply:
Deleted, thanks for spotting it.